src/ir/syntax.hpp

```
@@ -192,25 +192,50 @@ enum class CallKind {
+/// Identity of a helper/kfunc plus the diagnostics that travel with that identity.
+///
+/// The pair (func, kind) is the key: it uniquely identifies the target given a
+/// platform + program type. The remaining fields (name, is_supported,
+/// unsupported_reason) are a *snapshot* of what the platform lookup returned
+/// at unmarshal time; equality is by key only.
+///
+/// That means `CallTarget{.func=1}` and a fully-populated target for the same
+/// (func, kind) compare equal. See `CallContract` for the matching pattern
+/// on the semantic side. `Callx` and `CallBtf` are the key-only sibling
+/// instructions that resolve the contract on demand instead of caching it.
+struct CallTarget {
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: name, unsupported_reason

(cppcoreguidelines-pro-type-member-init)